lesson17/src/js/main.js

```diff
@@ -0,0 +1,225 @@
+function showLoadingImage() {
+  const main = document.getElementById("main");
+  const fragment = document.createDocumentFragment();
+  const renderLoadingBox = document.createElement("div");
+  renderLoadingBox.classList.add("loading");
+  renderLoadingBox.id = "js-loading";
+  const loadingImage = document.createElement("img");
+  loadingImage.src = "images/loading.gif";
+  loadingImage.width = 80;
+  loadingImage.height = 80;
+  loadingImage.classList.add("loading__image");
+  fragment.appendChild(renderLoadingBox).appendChild(loadingImage);
+  document.body.insertBefore(fragment, main);
+}
+
+function removeLoading() {
+  document.getElementById("js-loading").remove();
+}
+
+function displayErrorMassage(error) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const createTextBox = document.createElement("p");
+  createTextBox.classList.add("");
+  createTextBox.textContent = error;
+  slideImageArea.appendChild(createTextBox);
+}
+
+function renderSlideArea() {
+  const slideImageArea = document.createElement("div");
+  slideImageArea.classList.add("mainvisual__images__inner");
+
+  const sliderArea = document.querySelector(".mainvisual__images");
+
+  sliderArea.appendChild(slideImageArea);
+}
+
+function renderSlideImage(item) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+
+  const slider = document.createElement("ul");
+  slider.classList.add("slider");
+
+  const fragment = document.createDocumentFragment();
+  item.forEach((image, i) => {
+    const sliderItem = document.createElement("li");
+    sliderItem.classList.add("slider__item");
+    sliderItem.setAttribute("data-slide-index", i + 1);
+
+    const sliderImage = document.createElement("img");
+    sliderImage.classList.add("slider__image");
+    sliderImage.src = image.image;
+    sliderImage.width = image.width;
+    sliderImage.height = image.height;
+
+    if (i === 0) {
+      sliderItem.classList.add("is-active");
+    }
+
+    fragment.appendChild(sliderItem).appendChild(sliderImage);
+  });
+
+  slideImageArea.appendChild(slider).appendChild(fragment);
+}
+
+function renderSlideButton() {
+  let fragment = document.createDocumentFragment();
+
+  const buttonArea = document.createElement("div");
+  buttonArea.classList.add("slider__button");
+
+  const prevButton = document.createElement("button");
+  prevButton.setAttribute("class", "arrow prev");
```

Comment:
IMO
I found this one easier to understand.
```suggestion
  prevButton.classList.add("arrow", "prev");
```

Reply:
Thank you for your review! I didn't know that classList.add can also add multiple classes.
I modified it to the method you taught me.

[a11f6fb](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/a11f6fb37f4bb6e609f2928f3d26bee68df1de3c)